scripts/entity/base_entity.gd

```diff
@@ -15,13 +15,12 @@ class_name BaseEntity extends Node
 func _ready() -> void:
 
 	# if the movement component and input component exist connect the movement together.
-	if movement_component and input_component:
+	if movement_component and input_component and attack_component:
 		input_component.on_horizontal_movement_input.connect(movement_component.on_movement_input)
 		input_component.on_jump_input_started.connect(movement_component.on_jump_input_started)
 		input_component.on_jump_input.connect(movement_component.on_jump_input)
 		input_component.on_jump_input_cancelled.connect(movement_component.on_jump_input_cancelled)
-
-	if input_component and attack_component:
+		movement_component.on_direction_changed.connect(attack_component.update_direction_specific_items)
```

Comment:
I'd argue we could have stationary and need direction too. We have the idea of "turrets" that were mentioned under possible enemies, which would need to sit and face in a direction. I think this can prolly be ignored until they're introduced tho.